### PR TITLE
EGLUtils: allow selecting a high priority EGL context

### PR DESCRIPTION
**Pull request by lrusak**
I also had to change to use `CEGLAttributes` so we could add the `EGL_CONTEXT_PRIORITY_LEVEL_IMG`
 attribute easily.
```
    This extension allows an EGLContext to be created with a priority
    hint. It is possible that an implementation will not honour the
    hint, especially if there are constraints on the number of high
    priority contexts available in the system, or system policy limits
    access to high priority contexts to appropriate system privilege
    level. A query is provided to find the real priority level assigned
    to the context after creation.
```

see, https://www.khronos.org/registry/EGL/extensions/IMG/EGL_IMG_context_priority.txt

### CODE REVIEW

File: xbmc/utils/EGLUtils.h [hunk 1]
```diff
@@ -135,7 +135,7 @@ class CEGLContextUtils final
 
   bool CreateSurface(EGLNativeWindowType nativeWindow);
   bool CreatePlatformSurface(void* nativeWindow, EGLNativeWindowType nativeWindowLegacy);
-  bool CreateContext(const EGLint* contextAttribs);
+  bool CreateContext(CEGLAttributes<4> contextAttribs);
```
Comment (pkerling) [inline xbmc/utils/EGLUtils.h]: `const&`

Comment (pkerling) [reply xbmc/utils/EGLUtils.h]: OK, disregard :clown_face: wouldn't help much obviously

**Comment by pkerling** [comment]
`CEGLAttributes` was meant for local stack-allocated attribute lists. Clearly you have to jump through some hoops if you want to pass arbitrary attribute lists around, so it may not be the best tool for the job. Don't want to say that C pointers would be better ofc ;-)

If you still want to use it, at least make a shorthand type on `CEGLUtils` so the code doesn't need to be changed everywhere when the attribute count has to be changed.

Other options would be to change `CEGLAttributes` to heap allocation using a vector (at a small runtime cost) or make a separate class that uses a vector.

**Comment by lrusak** [comment]
@pkerling yes, that's a good idea 👍

File: xbmc/utils/EGLUtils.cpp [hunk 2]
```diff
@@ -17,6 +17,20 @@
 
 #include <EGL/eglext.h>
 
+namespace
+{
+// remove when Raspberry Pi updates their EGL headers
```
Comment (Rechi) [inline xbmc/utils/EGLUtils.cpp]: add `//! @todo` so it show up in the documentation

File: xbmc/utils/EGLUtils.cpp [hunk 3]
```diff
@@ -210,15 +224,29 @@ bool CEGLContextUtils::ChooseConfig(EGLint renderableType, EGLint visualId)
...
 
-bool CEGLContextUtils::CreateContext(const EGLint* contextAttribs)
+bool CEGLContextUtils::CreateContext(CEGLAttributesVec contextAttribs)
 {
   if (m_eglContext != EGL_NO_CONTEXT)
   {
     throw std::logic_error("Do not call CreateContext when context has already been created");
   }
 
+  if (CEGLUtils::HasExtension(m_eglDisplay, "EGL_IMG_context_priority"))
```
Comment (pkerling) [inline xbmc/utils/EGLUtils.cpp]: Why don't you `#ifdef EGL_IMG_context_priority` here instead of defining the stuff manually? Would look nicer imo.

Comment (lrusak) [reply xbmc/utils/EGLUtils.cpp]: I have mixed opinions because I would have to ifdef twice. It's up to you if you want me to do so 👍

**Comment by lrusak** [comment]
jenkins build this please

File: xbmc/utils/EGLUtils.cpp (hunk 3 above)
Comment (pkerling) [reply xbmc/utils/EGLUtils.cpp]: IMO it's cleaner, but I won't insist. It's marked for removal anyway.

**Comment by lrusak** [comment]
jenkins build and merge

**Pull request merged**